### PR TITLE
BucketAccessRequest field update

### PR DESCRIPTION
**Pull request by copejon**
This PR replaces:

- `bucketAccessRequest.spec.bucket` with `bucketAccessRequest.spec.bucketRequest`.  The Bucket should be dereferenced through the user's BucketRequest.

- `bucketAccessRequest.spec.bucketAccessRequestName` field with `bucketAccessRequest.spec.bucketAccessName` to reflect the relationship between the user and controller side APIs.

Edit to include:
Notable changes include:

- simplified controller-gen, result is CRDs are genereated to config/crd/, rather than config/crd/bases
- set path kb tags to force k8s DNS formatted object names. Else crds are generated with metadata.name: CamelCaseObject  which fails
- manually added approval annotation.  this is a pain as it's obliterated every time the CRDs are regenerated.  Will require a small custom tool.
- and the actual stated purpose of the PR: brought the go types in line with the KEP definitions

### CODE REVIEW

**Comment by copejon** [comment]
Running code generator produces:
```
$ ./hack/update-codegen.sh
API rule violation: list_type_missing,github.com/container-object-storage-interface/api/apis/cosi.sigs.k8s.io/v1alpha1,BucketAccessClass,SupportedProtocols
API rule violation: list_type_missing,github.com/container-object-storage-interface/api/apis/cosi.sigs.k8s.io/v1alpha1,BucketClass,AdditionalPermittedNamespaces
API rule violation: list_type_missing,github.com/container-object-storage-interface/api/apis/cosi.sigs.k8s.io/v1alpha1,BucketClass,AnonymousAccessModes
API rule violation: list_type_missing,github.com/container-object-storage-interface/api/apis/cosi.sigs.k8s.io/v1alpha1,BucketClass,SupportedProtocols
API rule violation: list_type_missing,github.com/container-object-storage-interface/api/apis/cosi.sigs.k8s.io/v1alpha1,BucketSpec,PermittedNamespaces
API rule violation: list_type_missing,github.com/container-object-storage-interface/api/apis/cosi.sigs.k8s.io/v1alpha1,PolicyActions,Allow
API rule violation: list_type_missing,github.com/container-object-storage-interface/api/apis/cosi.sigs.k8s.io/v1alpha1,PolicyActions,Deny
API rule violation: names_match,github.com/container-object-storage-interface/api/apis/cosi.sigs.k8s.io/v1alpha1,BucketAccessRequestSpec,BucketRequestName
```

**Comment by copejon** [comment]
dependent on #2 for API fixes

File: apis/cosi.sigs.k8s.io/v1alpha1/protocol_types.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ type ProtocolSignature string
 const (
 	ProtocolSignatureS3    ProtocolSignature = "s3"
 	ProtocolSignatureAzure ProtocolSignature = "azure"
-	ProtcolSignatureGCS    ProtocolSignature = "gcs"
```
Comment (wlan0) [inline apis/cosi.sigs.k8s.io/v1alpha1/protocol_types.go:8]: This is a nice find.

**Comment by wlan0** [comment]
Please remove the .DS_Store file and rebase. Otherwise LGTM

**Comment by wlan0** [comment]
LGTM

**Pull request merged**